packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1122,12 +1128,12 @@ export function attach(
   // When a mount or update is in progress, this value tracks the root that is being operated on.
   let currentRootID: number = -1;
 
-  function getReplayPersistentID(obj: Fiber) {
-    const id = __RECORD_REPLAY_ARGUMENTS__.getPersistentId(obj);
-    if (!id) {
-      throw new Error(
-        `Missing persistent ID for fiber ${obj} ${obj.constructor}`,
-      );
+  function getReplayPersistentID(fiber: Fiber) {
+    const id = __RECORD_REPLAY_ARGUMENTS__.getPersistentId(fiber);
+    if (id == null) {
+      const name = getDisplayNameForFiber(fiber) ?? '(Unknown)';
```

Comment:
:hand: `getDisplayNameForFiber` does call `getOrGenerateFiberID` inside, so we could end up with a stack overflow here, I think.  Could consider creating a JS object with a few bits of info extracted from the fiber (like I have in the routine eval logic) and stringify that, but I'd be okay if we just settled for `throw new Error("Missing persistent ID for fiber")` with no additional info if needed.

Reply:
Ah, good catch. Yes happy to make that change